internal/module/module.go

```diff
@@ -61,11 +53,60 @@ func parseModuleFile(repoRoot string) (*build.File, error) {
 // Collects the mapping of module names (e.g. "rules_go") to user-configured apparent names (e.g.
 // "my_rules_go"). See https://bazel.build/external/module#repository_names_and_strict_deps for more
 // information on apparent names.
-func collectApparentNames(m *build.File) map[string]string {
+func collectApparentNames(repoRoot, relPath string) (map[string]string, error) {
+	apparentNames := make(map[string]string)
+	seenFiles := make(map[string]struct{})
+	filesToProcess := []string{relPath}
+
+	for len(filesToProcess) > 0 {
+		f := filesToProcess[0]
+		filesToProcess = filesToProcess[1:]
+		if _, seen := seenFiles[f]; seen {
+			continue
+		}
+		seenFiles[f] = struct{}{}
+		bf, err := parseModuleSegment(repoRoot, f)
+		if err != nil {
+			if f == "MODULE.bazel" && os.IsNotExist(err) {
+				// If there is no MODULE.bazel file, return an empty map but no error.
+				// Languages will know to fall back to the WORKSPACE names of repos.
+				return nil, nil
+			}
+			return nil, err
+		}
+		names, includeLabels := collectApparentNamesAndIncludes(bf)
+		for name, apparentName := range names {
+			apparentNames[name] = apparentName
+		}
+		for _, includeLabel := range includeLabels {
+			var l label.Label
+			l, err = label.Parse(includeLabel)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse include label %q: %v", includeLabel, err)
+			}
```

Comment:
no need to declare the label
```suggestion
			l, err := label.Parse(includeLabel)
			if err != nil {
				return nil, fmt.Errorf("failed to parse include label %q: %v", includeLabel, err)
			}
```